src/ripple/protocol/impl/STTx.cpp

```diff
@@ -286,25 +295,32 @@ STTx::checkSingleSign () const
     if (isFieldPresent (sfSigners))
         return false;
 
-    bool ret = false;
+    bool validSig = false;
     try
     {
-        ECDSA const fullyCanonical = (getFlags() & tfFullyCanonicalSig)
-            ? ECDSA::strict
-            : ECDSA::not_strict;
-
-        RippleAddress n;
-        n.setAccountPublic (getFieldVL (sfSigningPubKey));
+        bool const fullyCanonical = (getFlags() & tfFullyCanonicalSig)
+            ? true
+            : false;
```

Comment:
No need to the trinary `? true : false` here, especially since the variable is explicitly `bool`.